### PR TITLE
Move Jest/RTL doc into main Stripes docs

### PR DESCRIPTION
**Pull request by ryandberger**
- Fulfills [STRIPES-768](https://issues.folio.org/browse/STRIPES-768)
- Trimmed down original document from https://github.com/folio-org/ui-plugin-eusage-reports/blob/master/doc/testing-with-jest-and-rtl.md
- Many thanks to @MikeTaylor for the original document

### CODE REVIEW

**Review by MikeTaylor — APPROVED** [review]
Nice to see this in a location where developers new to Jest/RTL are more likely to find it.

File: doc/unit-testing-with-jest-and-rtl.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	  }
+	]
+
+
+## Configuring Jenkins: running tests in CI
+
+Once you have your tests running, you want them to run automatically in CI. Assuming your project is set up in the standard way in the FOLIO Jenkins system, all you need to do is add these lines to your `Jenkinsfile`:
+
+	runTest = 'yes'
+	runTestOptions = '--coverage'
```
Comment (zburke) [inline doc/unit-testing-with-jest-and-rtl.md:105]: GitHub actions is the new standard way; Jenkins is the grandfathered way. Regardless, how about:
> For projects using Jenkins for CI, add these lines to `./Jenkinsfile` ...
> For projects using GitHub Actions for CI, add these lines to `./.github/workflows/build-npm.yml`:
```
env:
  YARN_TEST_OPTIONS: '--ci --coverage --colors'
  JEST_JUNIT_OUTPUT_DIR: 'artifacts/jest-junit'
  JEST_COVERAGE_REPORT_DIR: 'artifacts/coverage-jest/lcov-report/'
...
steps:
  - name: Run yarn test
    run: xvfb-run --server-args="-screen 0 1024x768x24" yarn test $YARN_TEST_OPTIONS JEST_COVERAGE_REPORT_DIR: 'artifacts/coverage-jest/lcov-report/'

  - name: Publish Jest unit test results
    uses: docker://ghcr.io/enricomi/publish-unit-test-result-action:v1
    if: always()
    with:
      github_token: ${{ github.token }}
      files: "${{ env.JEST_JUNIT_OUTPUT_DIR }}/*.xml"
      check_name: Jest Unit Test Results
      comment_mode: update last
      comment_title: Jest Unit Test Statistics
  
  - name: Publish Jest coverage report
    uses: actions/upload-artifact@v2
    if: always()
    with:
      name: jest-coverage-report
      path: ${{ env.JEST_COVERAGE_REPORT_DIR }}
      retention-days: 30
```

**Review by zburke — CHANGES_REQUESTED** [review]
I noticed a few minor typos that should be fixed and I think it may be better to use canonical URLs that reference specific commits rather than relative URLs that reference branches. 

The rest of my comments you can take or leave.

File: doc/unit-testing-with-jest-and-rtl.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+The CI configuration knows to find the coverasge analysis results in `artifacts/coverage-jest`, which is where the Jest configuration says to leave them, so coverage artifacts will appear in Jenkins.
```
[the remark this answers is not in the document]
Comment (ryandberger) [reply doc/unit-testing-with-jest-and-rtl.md:107]: Is this a typo?

[the remark this answers is not in the document]
Comment (ryandberger) [reply doc/unit-testing-with-jest-and-rtl.md]: Found the type in the doc. Taken care of.

**Review by zburke — APPROVED** [review]
LGTM!

**Review by mkuklis — APPROVED** [review]

**Pull request merged**